packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -357,12 +414,12 @@ class _SegmentedControlState<T extends Object> extends State<CupertinoSegmentedC
         cursor: kIsWeb ? SystemMouseCursors.click : MouseCursor.defer,
         child: GestureDetector(
           behavior: HitTestBehavior.opaque,
-          onTapDown: (TapDownDetails event) {
+          onTapDown: !widget.disabledChildren.contains(currentKey) ? (TapDownDetails event) {
```

Comment:
I think we can reverse the logic to avoid double negative. In style guide, we want to [avoid double negative in apis](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-double-negatives-in-apis), I think it's easier to understand for the logic here:)
Like: `widget.disabledChildren.contains(currentKey) ? null : (TapDownDetails event) {...}`